File: DailyChallenges/Scala/LeetCode_55.scala
object Solution {
    def canJump(nums: Array[Int]): Boolean = {
        var reach = 0
        for (i <- 0 until nums.length){
            if (i>reach) return false
            reach = Math.max(i+nums(i), reach)
        }
        true
    }
}

File: DailyChallenges/CPP/LeetCode_55.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int n = nums.size();
        vector<bool> visited(n, false); visited[0] = true;
        for (int i=0; i<n; ++i){
            if (visited[i])
                fill_n(visited.begin() + i + 1, min(n-i-1, nums[i]), true);
            if (visited[n-1]) return true;
        }
        return false;
    }
};
